### PR TITLE
Fixes BlockLiquidWrapper placing a Static fluid where it should place…

### PR DESCRIPTION
**Pull request by Iuhegi**
… a Flowing one.

I ran into this issue when attempting to create a block that places fluids in the world as they are pumped into it.

### CODE REVIEW

**Comment by liach** [comment]
Why?

**Comment by Iuhegi** [comment]
why what?

**Comment by diesieben07** [comment]
> Why?

Because otherwise the fluid will not start flowing.

File: src/main/java/net/minecraftforge/fluids/capability/wrappers/BlockLiquidWrapper.java [hunk 1]
```diff
@@ -34,8 +33,10 @@
 import net.minecraftforge.fluids.FluidRegistry;
 import net.minecraftforge.fluids.FluidStack;
 import net.minecraftforge.fluids.capability.FluidTankProperties;
-import net.minecraftforge.fluids.capability.IFluidTankProperties;
 import net.minecraftforge.fluids.capability.IFluidHandler;
+import net.minecraftforge.fluids.capability.IFluidTankProperties;
+
+import javax.annotation.Nullable;
```
Comment (diesieben07) [inline src/main/java/net/minecraftforge/fluids/capability/wrappers/BlockLiquidWrapper.java]: Don't reorganize imports.

Comment (Iuhegi) [reply src/main/java/net/minecraftforge/fluids/capability/wrappers/BlockLiquidWrapper.java]: hokay, lemme fix that

File: src/main/java/net/minecraftforge/fluids/capability/wrappers/BlockLiquidWrapper.java [hunk 2]
```diff
@@ -33,8 +34,8 @@
 import net.minecraftforge.fluids.Fluid;
 import net.minecraftforge.fluids.FluidRegistry;
 import net.minecraftforge.fluids.FluidStack;
-import net.minecraftforge.fluids.capability.FluidTankProperties;
 import net.minecraftforge.fluids.capability.IFluidTankProperties;
+import net.minecraftforge.fluids.capability.FluidTankProperties;
```
Comment (diesieben07) [inline src/main/java/net/minecraftforge/fluids/capability/wrappers/BlockLiquidWrapper.java]: Still reorganizing imports.

File: src/main/java/net/minecraftforge/fluids/capability/wrappers/BlockLiquidWrapper.java [hunk 3]
```diff
@@ -78,7 +79,7 @@ public int fill(FluidStack resource, boolean doFill)
         if (doFill)
         {
             Material material = blockLiquid.getDefaultState().getMaterial();
-            BlockLiquid block = BlockLiquid.getStaticBlock(material);
+            BlockDynamicLiquid block = BlockLiquid.getFlowingBlock(material);
```
Comment (diesieben07) [inline src/main/java/net/minecraftforge/fluids/capability/wrappers/BlockLiquidWrapper.java]: You don't need to change the type here.

**Review by diesieben07 — CHANGES_REQUESTED** [review]

File: src/main/java/net/minecraftforge/fluids/capability/wrappers/BlockLiquidWrapper.java (hunk 3 above)
Comment (Iuhegi) [reply src/main/java/net/minecraftforge/fluids/capability/wrappers/BlockLiquidWrapper.java]: ok

**Comment by Iuhegi** [comment]
What a scrub I am XD

**Review by diesieben07 — APPROVED** [review]

**Comment by LexManos** [comment]
You shouldn't need to pass in the flowing state.. you should set the source, and let it flow naturally in the next update tick...

**Comment by diesieben07** [comment]
> You shouldn't need to pass in the flowing state.. you should set the source, and let it flow naturally in the next update tick...

... which it doesn't do. Only the flowing thing actually updates when placed.

**Comment by KnightMiner** [comment]
I've noticed a similar issue when implementing a custom bucket. I had to hardcode a block update for vanilla fluids after placing them as otherwise they never starts flowing.

**Comment by Actuarius** [comment]
@tterrag1098 added labels [Bug]

**Comment by liach** [comment]
Wait @Actuarius is back?

**Comment by mezz** [comment]
Interestingly the code here is not used by forge buckets because they defer to the vanilla bucket for lava and water. Modded fluid containers or placers will use it though.
Checking the vanilla fluid placement code (`ItemBucket#tryPlaceContainedLiquid`), this PR is correct.

**Review by liach — APPROVED** [review]

**Pull request merged**